src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,74 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Class<?>> implementaionMap = new HashMap<>();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementatiomInstance = null;
+        Class<?> implementationClazz = findImplementation(interfaceClazz);
+        if (!implementationClazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed,"
+                    + " missing @Component annotation on the class "
+                    + implementationClazz.getName());
+        }
+        Field [] declaredField = implementationClazz.getDeclaredFields();
+        for (Field field : declaredField) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementatiomInstance = createNewInstance(implementationClazz);
```

Comment:
I believe we do not want our instance creation inside the loop, what do you think?